app/services/LocationService.js

```diff
@@ -212,10 +213,15 @@ export default class LocationServices {
     const locationData = new LocationData();
     // handles edge cases around Android where start might get called again even though
     // the service is already created.  Make sure the listeners are still bound and exit
-    if (isBackgroundGeolocationConfigured) {
-      BackgroundGeolocation.start();
-      return;
-    }
+    let result = BackgroundGeolocation.checkStatus(({ isRunning }) => {
+      if (isRunning) {
+        BackgroundGeolocation.start();
+        return true;
+      } else {
+        return false;
+      }
+    });
+    if (result) return;
```

Comment:
From what I can tell there is no return type of [checkStatus()](https://github.com/mauron85/react-native-background-geolocation#checkstatussuccess-fail). 

I think this will always be false, remove.